source/agora/consensus/protocol/Nominator.d

```diff
@@ -228,24 +228,33 @@ extern(D):
         checks that the UTXO key for enrollment exists and new SCP should be
         created because of new enrollment of this node.
 
+        Params:
+            height = block height for creating SCP
+
     ***************************************************************************/
 
-    public void createSCPObject () nothrow @safe
+    public void createSCPObject (in Height height) nothrow @safe
     {
+        import scpd.types.Stellar_types;
         try
         {
             // TODO: We should apply the situation where this validator has
             // enrolled with another UTXO after its previous enrollment expired.
-            auto self_enroll = this.enroll_man.getEnrollmentKey();
-            if (this.scp is null && self_enroll != Hash.init)
+            const node_id = this.getSelfUTXOIndex(height + 1);
```

Comment:
Looking at the call chain, I think the caller of `createSCPObject` already has all the infos needed.

Reply:
I don't think we need `getSelfUTXOIndex` with your comment. I removed it.